### DISCUSSION TITLE
Let's talk about CAN-FD, Autosar,  'n stuff!

### DISCUSSION THREAD

**Discussion by christoph2** [christoph2/pyxcp#73] [Ideas]
Any questions or ideas?  :-)

**Comment by raragon71** [comment]
I'm trying to automate some regression/integration test I need to do on an ECU, and although I'm not planning to inject or read anything using XCP (for now), I do need to simulate a full load environment (my direct 'customers' are system engineers and they heavily use XCP for all their calibration and algorithm adjustments). So I need to set up some DAQ lists and verify the integrity and periodicity of the data being read, mainly because they keep raising defects to the XCP driver (legacy, custom made so not the best code around) but normally half of the times the issues are more related to THEIR CANape configuration... So I'd like to test XCP independently from that tool, to make sure my XCP driver works as expected under their regular use case... 
So I see a lot of potential on your PyXCP code, especially because I already have a Python-can based custom test script which I keep adding test cases and it's running on RasPi (although I might need to eventually switch to an Intel NUC)... so I'd like to give PyXCP (and PyA2L) a shot to see how far can I go for automating the XCP ddriver and not needing to use/rely on CANape...

**Comment by ankur1717** [comment]
Hi Chris
I'm new to PCAN CCP, Im trying hardware and PC interface. I'm bit confused regarding what master tool i have to use to communicate with my hardware. My hardware is based on C2000 series from TI and I'm using PCAN CCP protocol.
